org.eclipse.scanning.api/src/org/eclipse/scanning/api/malcolm/IMalcolmDevice.java

```diff
@@ -57,10 +57,20 @@
  * <img src="./doc/device_state.png" />
 
  * @author Matthew Gerring
+ * @param <T> 
  *
  */
 public interface IMalcolmDevice<T> extends IRunnableEventDevice<T>, IMalcolmEventPublisher, ILatchableDevice, IAttributableDevice, IValidator<T> {
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png 'Severity: CRITICAL') Move constants to a class or enum. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1214)